File: backend/myapp/views/index/operations.py
from .face_processor import FaceProcessor
from rest_framework.decorators import api_view
from myapp.utils import image_to_base64, base64_to_image
from myapp.handler import APIResponse
predictor_path = 'shape_predictor_68_face_landmarks.dat'
processor = FaceProcessor(predictor_path)


@api_view(['POST'])
def thin(request):
    image = request.data['image']
    if(image):
        res = image_to_base64(processor.face_thin(base64_to_image(image)))
        return APIResponse(code=0, msg='操作成功', data={'image': res})
    return APIResponse(code=0, msg='操作失败')

@api_view(['POST'])
def bigeye(request):
    image = request.data['image']
    if(image):
        res = image_to_base64(processor.face_bigeye(base64_to_image(image)))
        return APIResponse(code=0, msg='操作成功', data={'image': res})
    return APIResponse(code=0, msg='操作失败')


@api_view(['POST'])
def smooth(request):
    image = request.data['image']
    if(image):
        res = image_to_base64(processor.face_smooth(base64_to_image(image)))
        return APIResponse(code=0, msg='操作成功', data={'image': res})
    return APIResponse(code=0, msg='操作失败')

@api_view(['POST'])
def whiten(request):
    image = request.data['image']
    if(image):
        res = image_to_base64(processor.face_whiten(base64_to_image(image)))
        return APIResponse(code=0, msg='操作成功', data={'image': res})
    return APIResponse(code=0, msg='操作失败')

File: frontend/src/api/index/operations.js
/**
 * api
 */
import axios from '@/utils/request.js'

const api = {
  smoothApi:'/myapp/index/operations/smooth',
  thinApi:'/myapp/index/operations/thin',
  bigeyeApi:'/myapp/index/operations/bigeye',
  whitenApi:'/myapp/index/operations/whiten'
}

export const smoothApi = function (data) {
  return axios({
    url: api.smoothApi,
    method: 'post',
    data: data
  })
}
export const thinApi = function (data) {
  return axios({
    url: api.thinApi,
    method: 'post',
    data: data
  })
}
export const bigeyeApi = function (data) {
  return axios({
    url: api.bigeyeApi,
    method: 'post',
    data: data
  })
}
export const whitenApi = function (data) {
  return axios({
    url: api.whitenApi,
    method: 'post',
    data: data
  })
}

